### PR TITLE
Improve logging and error handling

### PR DESCRIPTION
**Pull request by gnardini**
Changes:
- Pretty Print is not used in production anymore, just the json logger.
- Added retrying for Blockscout queries.
- Moved to a slightly larger instance class (we were in the smallest one and I saw a couple of memory-related errors).
- Revised the severity of logs.
- Made sure all logs were in json format.

I tested all this locally and deployed in alfajores

- Fixes [#1882](https://github.com/valora-inc/wallet/issues/1882)

### CODE REVIEW

File: src/helpers/Input.ts [hunk 1]
```diff
@@ -31,8 +31,11 @@ export class Input {
 
     try {
       return coder.decodeParameters(abi, this.data)
-    } catch (e) {
-      logger.debug(`Error decoding input: ${(e as Error).message}`)
+    } catch (error) {
+      logger.warn({
+        type: 'INPUT_DECODE_ERROR',
+        error: (error as Error)?.message,
```
Comment (silasbw) [inline src/helpers/Input.ts:37]: I think you can include the error directly (vs `error.message`): https://github.com/trentm/node-bunyan#log-method-api

File: src/logger.ts [hunk 2]
```diff
@@ -2,13 +2,14 @@ import * as Logger from 'bunyan'
...
 
-const consoleStream = new PrettyStream()
-consoleStream.pipe(process.stdout)
-
-const streams: Logger.Stream[] = [{ stream: consoleStream, level: 'info' }]
+const streams: Logger.Stream[] = []
 if (process.env['GAE_APPLICATION']) {
   const loggingBunyan = new LoggingBunyan()
   streams.push(loggingBunyan.stream('info'))
+} else {
```
Comment (silasbw) [inline src/logger.ts:9]: Great!  I've been annoyed at the pretty printing logs showing up in google cloud :pray:

File: src/utils.ts [hunk 3]
```diff
@@ -145,3 +146,26 @@ export async function getContractKit(): Promise<ContractKit> {
     throw new Error('Failed to create contractKit instance')
   }
 }
+
+export async function runWithRetries<ReturnType>(
```
Comment (silasbw) [inline src/utils.ts:150]: should you move this to another PR?  i think it's worth more discussion and log improvements all look good.

File: src/blockscout.ts [hunk 4]
```diff
@@ -198,12 +203,16 @@ export class BlockscoutAPI extends RESTDataSource {
     return transactions
   }
 
+  async queryBlockscoutWithRetry(body: Body) {
```
[the remark this answers is not in the document]
Comment (gnardini) [reply src/blockscout.ts:206]: It fails very often, so I figured it would be better to retry here already to avoid the round trips.

File: src/helpers/Input.ts (hunk 1 above)
Comment (gnardini) [reply src/helpers/Input.ts:37]: Right, I figured out after changing this that is was the pretty logs that were causing the untidy logs, I thought it was this, I'll revert these changes

File: src/utils.ts (hunk 3 above)
Comment (gnardini) [reply src/utils.ts]: Removed the retry from here. Blockscout queries fail _very_ often though so I figured we'd be doing the client a favor by retrying here, but I can get more data before coming back with this.

File: src/helpers/Input.ts (hunk 1 above)
Comment (gnardini) [reply src/helpers/Input.ts]: Added the error serializer thing :)

File: src/resolvers.ts [hunk 5]
```diff
@@ -219,7 +240,11 @@ export const resolvers = {
         })
         return { rate: rate.toNumber() }
       } catch (error) {
-        logger.error(error, { type: 'ERROR_CURRENCY_CONVERSION' })
+        logger.error({
+          error,
+          type: 'CURRENCY_CONVERSION_ERROR',
+          sourceCurrencyCode: args.sourceCurrencyCode,
```
Comment (dievazqu) [inline src/resolvers.ts:246]: I think it would be helpful to print all the args, specially the timestamp.

Comment (dievazqu) [reply src/resolvers.ts:246]: Also, maybe this is should be a `warn` instead of an `error`

Comment (gnardini) [reply src/resolvers.ts:246]: Hmm, I think it's an error, this is called directly from Valora and the app is quite limited without it since we can't show local amounts. At least it's something that we should track how often it fails.

Comment (gnardini) [reply src/resolvers.ts:246]: Good catch on logging all the args!
